buildpacks/nodejs-corepack/src/main.rs

```diff
@@ -37,60 +40,76 @@ impl Buildpack for CorepackBuildpack {
-                            BuildPlanBuilder::new()
-                                .requires("node")
-                                .requires(&pkg_mgr)
-                                .provides(pkg_mgr)
-                                .build(),
-                        )
-                        .build()
-                },
-            )
-        } else {
-            DetectResultBuilder::fail().build()
-        }
+        let tracer = init_tracer(format!("{}", context.buildpack_descriptor.buildpack.id));
+        tracer.in_span("nodejs-corepack-detect", |_cx| {
+            let pkg_json_path = context.app_dir.join("package.json");
+            if pkg_json_path.exists() {
+                let pkg_json = PackageJson::read(pkg_json_path)
+                    .map_err(CorepackBuildpackError::PackageJson)?;
+                cfg::get_supported_package_manager(&pkg_json).map_or_else(
+                    || DetectResultBuilder::fail().build(),
+                    |pkg_mgr| {
+                        DetectResultBuilder::pass()
+                            .build_plan(
+                                BuildPlanBuilder::new()
+                                    .requires("node")
+                                    .requires(&pkg_mgr)
+                                    .provides(pkg_mgr)
+                                    .build(),
+                            )
+                            .build()
+                    },
+                )
+            } else {
+                DetectResultBuilder::fail().build()
+            }
+        })
     }
 
     fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
-        let pkg_mgr = PackageJson::read(context.app_dir.join("package.json"))
-            .map_err(CorepackBuildpackError::PackageJson)?
-            .package_manager
-            .ok_or(CorepackBuildpackError::PackageManagerMissing)?;
-
-        let env = &Env::from_current();
-
-        let corepack_version =
-            cmd::corepack_version(env).map_err(CorepackBuildpackError::CorepackVersion)?;
-
-        log_header(format!(
-            "Installing {} {} via corepack {corepack_version}",
-            pkg_mgr.name, pkg_mgr.version
-        ));
-
-        let shims_layer =
-            context.handle_layer(layer_name!("shim"), ShimLayer { corepack_version })?;
-        cmd::corepack_enable(&pkg_mgr.name, &shims_layer.path.join("bin"), env)
-            .map_err(CorepackBuildpackError::CorepackEnable)?;
-
-        let mgr_layer = context.handle_layer(
-            layer_name!("mgr"),
-            ManagerLayer {
-                package_manager: pkg_mgr,
-            },
-        )?;
-        let mgr_env = mgr_layer.env.apply(Scope::Build, env);
-        cmd::corepack_prepare(&mgr_env).map_err(CorepackBuildpackError::CorepackPrepare)?;
-
-        BuildResultBuilder::new().build()
+        let tracer = init_tracer(format!("{}", context.buildpack_descriptor.buildpack.id));
```

Comment:
will this clobber the tracer created during `detect`? or does that file get consumed before this happens?

Reply:
`detect` and `build` are different binaries. They shouldn't be in scope at the same time.